Parser/Node.cs

```diff
@@ -48,6 +59,30 @@ public record NWriteStmt (bool NewLine, NExpr[] Exprs) : NStmt {
 public record NAssignStmt (Token Name, NExpr Expr) : NStmt {
    public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
 }
+
+public record NReadStmt (Token[] List) : NStmt {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}
+
+public record NCallStmt (Token Name, NExpr[] Expr) : NStmt {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}
+
+public record NIfStmt (NExpr Expr, NStmt[] Stmts) : NStmt {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}
+
+public record NWhileStmt (NExpr Expr, NStmt[] Stmts) : NStmt {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}

```

Comment:
The body of a while loop is a _single_ statement. Even if there are multiple operations to complete, it is still a single CompoundStmt. 
`public record NWhileStmt (NExpr Condition, NStmt Body)`